airbyte-ci/connectors/pipelines/pipelines/gradle.py

```diff
@@ -50,66 +53,107 @@ def build_include(self) -> List[str]:
+        ] + self.build_include
+
+        yum_packages_to_install = [
+            "docker",  # required by :integrationTestJava.
+            "findutils",  # gradle requires xargs, which is shipped in findutils.
+            "jq",  # required by :airbyte-connector-test-harnesses:acceptance-test-harness to inspect docker images.
+            "npm",  # required by :format.
+            "pip",  # required by :format.
+            "rsync",  # required for gradle cache synchronization.
+        ]
+
+        # Define a gradle container which will be cached and re-used for all tasks.
+        # We should do our best to cram any generic & expensive layers in here.
+        gradle_container = (
+            self.dagger_client.container()
+            # Use a linux+jdk base image with long-term support, such as amazoncorretto.
+            .from_(f"amazoncorretto:{AMAZONCORRETTO_TAG}")
+            # Install a bunch of packages as early as possible.
+            .with_exec(
+                sh_dash_c(
+                    [
+                        # Update first, but in the same .with_exec step as the package installation.
+                        # Otherwise, we risk caching stale package URLs.
+                        "yum update -y",
+                        f"yum install -y {' '.join(yum_packages_to_install)}",
+                        # Deliberately soft-remove docker, so that the `docker` CLI is unavailable by default.
+                        # This is a defensive choice to enforce the expectation that, as a general rule, gradle tasks do not rely on docker.
+                        "yum remove -y --noautoremove docker",  # remove docker package but not its dependencies
+                        "yum install -y --downloadonly docker",  # have docker package in place for quick install
+                        # Remove any dangly bits.
+                        "yum clean all",
+                    ]
+                )
+            )
+            # Set GRADLE_HOME and GRADLE_USER_HOME to the directory which will be rsync-ed with the gradle cache volume.
+            .with_env_variable("GRADLE_HOME", "/root/.gradle")
+            .with_env_variable("GRADLE_USER_HOME", "/root/.gradle")
+            # Set RUN_IN_AIRBYTE_CI to tell gradle how to configure its build cache.
+            # This is consumed by settings.gradle in the repo root.
+            .with_env_variable("RUN_IN_AIRBYTE_CI", "1")
+            # Mount the gradle cache volume.
+            # We deliberately don't mount it at $GRADLE_HOME, instead we load it there and store it from there using rsync.
+            # This is because the volume is accessed concurrently by all GradleTask instances.
+            # Hence, why we synchronize the writes by setting the `sharing` parameter to LOCKED.
+            .with_mounted_cache("/root/gradle-cache", self.connector_java_build_cache, sharing=CacheSharingMode.LOCKED)
+            # Mount the parts of the repo which interest us in /airbyte.
```

Comment:
Awesome commenting work there, thank you for doing a better job than I did!

Reply:
Thanks, but it's easier to do for someone who rewrites something already existing, rather than having to come up with it in the first place over many successive iterations.